enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,315 @@
+	// Default is 6081
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayConfig configures an additional overlay network for peers that are
+	// not using OVN.
+	// +optional
+	HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+	// ipsecConfig enables and configures IPsec for pods on the pod network within the
+	// cluster.
+	// +optional
+	IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
+  // aclLoggingRateLimit allows for the configuring of max acl logging rate 
+  // +optional
+  aclLoggingRateLimit  *uint32 `json:"aclLoggingRateLimit,omitempty"`
+
+```
+
+The flag `aclLoggingRateLimit` defaults to `20`. 
+
+This will result in an updated OVNKubernetes Operator configuration object: (.spec.defaultNetwork.ovnKubernetesConfig) which will enable cluster-wide configuration of acl logging at cluster installation time: 
+
+```go
+spec:
+  defaultNetwork:
+    type: OVNKubernetes
+    ovnKubernetesConfig:
+      mtu: 1400
+      genevePort: 6081
+      ipsecConfig: {}
+      aclLoggingRateLimit: 20
+```
+
+### User Stories
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+Include a story on how this proposal will be operationalized:  lifecycled, monitored and remediated at scale.
+
+#### Story 1
+
+As an Openshift user I want to see what Network Policies and accompanied ACLs are dropping the most traffic in a cluster. For example, when a packet runs into a `drop` ACL it will log a message similar to the following 
+
+```
+2021-01-05T17:34:02.675Z|00004|acl_log(ovn_pinctrl0)|INFO|name="<Network Policy Name>", verdict=drop, severity=info: icmp,vlan_tci=0x0000,dl_src=50:54:00:00:00:02,dl_dst=50:54:00:00:00:01,nw_src=192.168.0.3,nw_dst=192.168.0.2,nw_tos=0,nw_ecn=0,nw_ttl=64,icmp_type=8,icmp_code=0
+```
+
+#### Story 2
+
+As an Openshift user I want to control the rate of ACL logging to ensure I can extract the useful information invloving accept/reject actions in high hit rate scenarios. 
+
+#### Story 3 
+
+As an Openshift user I want to monitor in realtime traffic that is hitting any network policies I have implemented. 
+
+--- TBD in implementation (Realtime Monitering would be nice here)
+
+### Implementation Details/Notes/Constraints 
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+The following codebase changes will need to done before this feature becomes usable  
+
+- OVN Kubernetes 
+
+  - Move from a cluster wide namespace default deny port group to namespace specific one 
+  - Add logging flag and custom OVN meter to all created ACLs, see below 
```

Comment:
-"... to all created ACLs, see below" ->  "... to all created ACLs. See below see below for an example of additional logging needed:"
- How many spots in the code will need this? Ballpark: 2, 50, 100, more?

Reply:
The Upstream work has merged -> https://github.com/ovn-org/ovn-kubernetes/pull/1654 The change was only needed in around 10 places in the code base